WalletWasabi.Tests/Helpers/ArenaRequestHandlerAdapter.cs

```diff
@@ -0,0 +1,40 @@
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using WalletWasabi.WabiSabi.Backend.PostRequests;
+using WalletWasabi.WabiSabi.Backend.Rounds;
+using WalletWasabi.WabiSabi.Models;
+
+namespace WalletWasabi.Tests.Helpers
+{
+	public class ArenaRequestHandlerAdapter : IWabiSabiApiRequestHandler
```

Comment:
Seems like this is required because `ArenaRequestHandler` doesn't give its cancellation tokens to `Arena`, but I can't think of a good reason why that's the case.

For the purpose of this pull request this is of course the right thing to do but maybe in a followup we should just add a cancellation token arg to each method in `Arena`, and actually use it? While it holds the lock its operations are largely CPU bound, so right now passing these won't do much, but in the long run we probably need to move away from that approach, and this PR already takes us a small step in that direction in the way it resolves outputs to coins.

Reply:
I can do that in a different PR.